packages/amplication-design-system/src/components/DataGrid/DataGrid.tsx

```diff
@@ -0,0 +1,261 @@
+import React, { useCallback, ReactNode, useMemo } from "react";
+import {
+  DataTable,
+  DataTableContent,
+  DataTableHead,
+  DataTableRow,
+  DataTableHeadCell,
+  DataTableBody,
+} from "@rmwc/data-table";
+import "@rmwc/data-table/styles";
+import { Icon } from "@rmwc/icon";
+import classNames from "classnames";
+import { keyBy } from "lodash";
+
+import SearchField from "../SearchField/SearchField";
+import {
+  SelectMenu,
+  SelectMenuModal,
+  SelectMenuItem,
+  SelectMenuList,
+} from "../SelectMenu/SelectMenu";
+import { EnumButtonStyle } from "../Button";
+import "./DataGrid.scss";
+
+export type SortData = {
+  field: string | null;
+  order: SortOrder | null;
+};
+
+export enum SortOrder {
+  Asc = 0,
+  Desc = 1,
+}
+
+export enum EnumTitleType {
+  PageTitle = "PageTitle",
+  SectionTitle = "SectionTitle",
+}
+
+type FilterItem = {
+  value: string;
+  label: string;
+};
+
+type FilterChangeData = {
+  filterName: string;
+  value: string;
+};
+
+export type DataFilter = {
+  name: string;
+  title: string;
+  selected: Set<string>;
+  filterItems: FilterItem[];
+};
+
+export type DataField = {
+  name: string;
+  title: string;
+  icon?: string;
+  sortable?: boolean;
+  minWidth?: boolean;
+};
+
+export type Props = {
+  /** Fields to display in the data grid */
+  fields: DataField[];
+  /** Title of the data grid */
+  title: string;
+  /** The type of presentation for the title of the data grid */
+  titleType?: EnumTitleType;
+  /** Whether the data grid is in a state of loading */
+  loading: boolean;
+  /** The field the data is ordered by */
+  sortDir: SortData;
+  children: ReactNode;
+  /** Optional elements to present before the data grid's toolbar */
+  toolbarContentStart?: ReactNode;
+  /** Optional elements to present after the data grid's toolbar */
+  toolbarContentEnd?: ReactNode;
+  /** The conditions the data is filtered by */
+  filters?: DataFilter[];
+  onSortChange?: (fieldName: string, order: number | null) => void;
```

Comment:
Better pass SortData instead of two arguments to have a consistency with the sortData argument